components/api-resource-mgt/org.wso2.carbon.identity.api.resource.collection.mgt/src/main/java/org/wso2/carbon/identity/api/resource/collection/mgt/listener/ConsoleRoleListener.java

```diff
@@ -0,0 +1,256 @@
+import org.wso2.carbon.identity.api.resource.collection.mgt.model.APIResourceCollection;
+import org.wso2.carbon.identity.api.resource.collection.mgt.model.APIResourceCollectionSearchResult;
+import org.wso2.carbon.identity.api.resource.mgt.APIResourceMgtException;
+import org.wso2.carbon.identity.application.common.model.Scope;
+import org.wso2.carbon.identity.role.v2.mgt.core.RoleConstants;
+import org.wso2.carbon.identity.role.v2.mgt.core.RoleManagementService;
+import org.wso2.carbon.identity.role.v2.mgt.core.exception.IdentityRoleManagementException;
+import org.wso2.carbon.identity.role.v2.mgt.core.listener.AbstractRoleManagementListener;
+import org.wso2.carbon.identity.role.v2.mgt.core.model.Permission;
+import org.wso2.carbon.identity.role.v2.mgt.core.model.Role;
+import org.wso2.carbon.identity.role.v2.mgt.core.model.RoleBasicInfo;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static org.wso2.carbon.identity.api.resource.collection.mgt.constant.APIResourceCollectionManagementConstants.APIResourceCollectionConfigBuilderConstants.EDIT_ACTION;
+import static org.wso2.carbon.identity.api.resource.collection.mgt.constant.APIResourceCollectionManagementConstants.APIResourceCollectionConfigBuilderConstants.VIEW_ACTION;
+import static org.wso2.carbon.identity.role.v2.mgt.core.RoleConstants.APPLICATION;
+import static org.wso2.carbon.identity.role.v2.mgt.core.RoleConstants.CONSOLE_APP_AUDIENCE_NAME;
+import static org.wso2.carbon.identity.role.v2.mgt.core.RoleConstants.CONSOLE_ORG_SCOPE_PREFIX;
+import static org.wso2.carbon.identity.role.v2.mgt.core.RoleConstants.CONSOLE_SCOPE_PREFIX;
+
+/**
+ * Console role listener to populate organization console application roles permissions.
+ */
+public class ConsoleRoleListener extends AbstractRoleManagementListener {
+
+    @Override
+    public int getDefaultOrderId() {
+
+        return 4;
+    }
+
+    @Override
+    public boolean isEnable() {
+
+        return true;
+    }
+
+    @Override
+    public void preAddRole(String roleName, List<String> userList, List<String> groupList, List<Permission> permissions,
+                           String audience, String audienceId, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        if (APPLICATION.equals(audience)) {
+            List<Permission> consoleFeaturePermissions = getConsoleFeaturePermissions(permissions);
+            if (ArrayUtils.isNotEmpty(consoleFeaturePermissions.toArray())) {
+                // If console features are added to the role, then we need to we only need to persist the console
+                // permissions.
+                permissions.retainAll(consoleFeaturePermissions);
+            }
+        }
+    }
+
+    @Override
+    public void postGetRole(Role role, String roleId, String tenantDomain) throws IdentityRoleManagementException {
+
+
+        if (isConsoleRole(roleId, tenantDomain)) {
```

Comment:
Instead of using common method, let's write convenient logics.

if (!"CONSOLE".equals()role.getAudienceName() || !"application".equalsIgnoreCase(role.getAudience())) {
  return;
}